python/cuml/tests/test_base.py

```diff
@@ -327,3 +331,74 @@ def test_regressor_predict_dtype(cls):
     # in sklearn here: https://github.com/scikit-learn/scikit-learn/issues/22682
     y_pred = cls().fit(X32, y32).predict(X32)
     assert y_pred.dtype == np.float32
+
+
+@pytest.mark.parametrize(
+    "cls, kwargs",
+    [
+        (cuml.LogisticRegression, None),
+        (cuml.RandomForestClassifier, None),
+        (cuml.SVC, None),
+        (cuml.SVC, {"probability": True}),
+        (cuml.LinearSVC, None),
+        (cuml.KNeighborsClassifier, None),
+    ],
+)
+@pytest.mark.parametrize(
+    "target_kind", ["binary", "multiclass", "multitarget"]
+)
+@pytest.mark.parametrize("dtype_kind", ["int-monotonic", "int", "string"])
+def test_classifier_label_types(cls, kwargs, target_kind, dtype_kind):
+    supports_multitarget = [cuml.KNeighborsClassifier]
+    if target_kind == "multitarget" and cls not in supports_multitarget:
+        pytest.skip(f"{cls.__name__} doesn't support multitarget y")
+
+    labels = {
+        "int-monotonic": [0, 1, 2, 3],
+        "int": [5, 10, 15, 20],
+        "string": ["a", "b", "c", "d"],
+    }[dtype_kind]
+
+    if target_kind == "binary":
+        X, y = make_classification(n_samples=200, random_state=42, n_classes=2)
+        y = np.array(labels).take(y)
+    elif target_kind == "multiclass":
+        X, y = make_classification(
+            n_samples=200, random_state=42, n_classes=4, n_informative=4
+        )
+        y = np.array(labels).take(y)
+    elif target_kind == "multitarget":
+        X, y = make_multilabel_classification(
+            n_samples=200, random_state=42, n_classes=4
+        )
+        y = np.array(labels).take(y)
```

Comment:
`make_multilabel_classification` produces a binary matrix so only the first two labels are ever used.